lms/static/js/learner_dashboard/views/course_entitlement_view.js

```diff
@@ -0,0 +1,300 @@
+    'use strict';
+
+    define(['backbone',
+        'jquery',
+        'underscore',
+        'gettext',
+        'moment',
+        'edx-ui-toolkit/js/utils/html-utils',
+        'js/learner_dashboard/models/course_entitlement_model',
+        'js/learner_dashboard/models/course_card_model',
+        'text!../../../templates/learner_dashboard/course_entitlement.underscore'
+    ],
+         function(
+             Backbone,
+             $,
+             _,
+             gettext,
+             moment,
+             HtmlUtils,
+             EntitlementModel,
+             CourseCardModel,
+             pageTpl
+         ) {
+             return Backbone.View.extend({
+                 tpl: HtmlUtils.template(pageTpl),
+
+                 events: {
+                     'change .session-select': 'updateEnrollBtn',
+                     'click .enroll-btn': 'handleEnrollChange',
+                     'click .popover-dismiss': 'hideVerificationDialog',
+                     'keydown .final-confirmation-btn': 'handleVerificationPopoverA11y'
+                 },
+
+                 initialize: function(options) {
+                     this.$el = options.$el;
+
+                     // Set up models and reload view on change
+                     this.courseCardModel = new CourseCardModel();
+                     this.entitlementModel = new EntitlementModel({
+                         availableSessions: this.formatDates(JSON.parse(options.availableSessions)),
+                         entitlementUUID: options.entitlementUUID,
+                         currentSessionId: options.currentSessionId,
+                         userId: options.userId
+                     });
+                     this.listenTo(this.entitlementModel, 'change', this.render);
+
+                     // Grab URLs that handle changing of enrollment and entering a newly enrolled session.
+                     this.enrollUrl = options.enrollUrl;
+                     this.courseHomeUrl = options.courseHomeUrl;
+
+                     // Grab elements from the parent card that work with this view and bind associated events
+                     this.$triggerOpenBtn = options.$triggerOpenBtn; // Opens and closes session selection view
+                     this.$dateDisplayField = options.$dateDisplayField; // Displays current session dates
+                     this.$enterCourseBtn = options.$enterCourseBtn; // Button link to course home page
+                     this.$courseCardMessages = options.$courseCardMessages; // Additional session messages
+                     this.$courseTitleLink = options.$courseTitleLink; // Title link to course home page
+                     this.$courseImageLink = options.$courseImageLink; // Image link to course home page
+                     this.$courseMessages = options.$courseMessages; // Extra messages attached to course Card
+                     this.$triggerOpenBtn.on('click', this.toggleSessionSelectionPanel.bind(this));
+
```

Comment:
For any reviewers - in case this part doesn't make sense, this is a way to link the view to its external card so that the view can be used in both the course and program dashboards.